LE Audio: Fix and add missing debug Kconfigs

Thalley opened the pull request:
Adds a few missing and fixes remaining debug kconfigs after https://github.com/zephyrproject-rtos/zephyr/pull/51897 was merged without them.

Thalley committed:
Bluetooth: Audio: Fix CAP Acceptor Debug log level

The implementation did not use the defined log level
Kconfig.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Fix CAP initiator debug level

Add missing debug level Kconfig option and use it.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Add missing codec debug Kconfig

Add misisng codec debug Kconfig for the codec.c file.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley commented:
Can't add you as reviewer, but ping @theob-pro

theob-pro commented on subsys/bluetooth/audio/Kconfig.baps:
```diff
@@ -273,6 +273,10 @@ legacy-debug-sym = BT_AUDIO_DEBUG_STREAM
 
+module = BT_AUDIO_CODEC
+module-str = "Bluetooth Audio Codec"
+source "subsys/bluetooth/common/Kconfig.template.log_config_bt"
```
You should not use this template if you don't want to provide the `legacy-debug-sym` argument. This template assumes you need compatibility with the old Bluetooth logging config variables. I suggest we recreate the kconfig template for "inherit" logging functionality, without the backwards compatibility and use it here.

Thalley replied:
I am not 100% sure I follow. I saw your other PR (https://github.com/zephyrproject-rtos/zephyr/pull/52608) to add the warning, which makes sense, but I am unsure what you want me to change here. 

How should the BT module define new logging modules as of right now, if this is incorrect? You mention recreating the Kconfig template, but will that be a different template?

theob-pro replied:
Yes it will be a new Kconfig template. You could use this: https://github.com/theob-pro/zephyr/blob/9a178776008dc3345a674e61371a83a5c3954d50/subsys/logging/Kconfig.template.log_config_inherit
We added it in the first place but removed it because of the backward compatibility need.

Thalley replied:
So currently there is no way of adding a new log level for a module? 

I'd not not introduce that in this PR, and since it sounds like you have a plan of what needs to be done, could I bother you to create the PR to add such a template?

theob-pro replied:
> So currently there is no way of adding a new log level for a module?
>
> I'd not not introduce that in this PR, and since it sounds like you have a plan of what needs to be done, could I bother you to create the PR to add such a template?

I will make a PR and you will be able to rebase on it if it get merged.

Thalley replied:
I can see that something like OTS simply does this:
```
module = BT_OTS
module-str = BT_OTS
source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config"
```

I should be able to just use that, right?

theob-pro replied:
Now that https://github.com/zephyrproject-rtos/zephyr/pull/52635 has been merged, you could use `source "subsys/logging/Kconfig.template.log_config_inherit"`. You just need to define the `parent-module` before the source. 

Here is how you could do it:
```
parent-module = BT
module = BT_AUDIO_CODEC
module-str = "Bluetooth Audio Codec"
source "subsys/logging/Kconfig.template.log_config_inherit"
```

Thalley replied:
@theob-pro Is that something we should generally do/suggest? 

By adding a `parent-module`, e.g. `CONFIG_BT_DEBUG`, we can basically use `CONFIG_BT_DEBUG` to disable all the children logs, right? This clearly had advantages, but wouldn't it also make it harder to enable e.g. a specific BT module? 

If I set `CONFIG_BT_DEBUG` as the parent, and it is not enabled, and I enable one of the children (e.g. `CONFIG_BT_AUDIO_CODEC_LOG_LEVEL_DBG`), do I get a warning or something, or is my `CONFIG_BT_AUDIO_CODEC_LOG_LEVEL_DBG` just silently ignored?

theob-pro replied:
No, the `CONFIG_BT_AUDIO_CODEC_LOG_LEVEL_DBG` will get the value of its parent only if `BT_AUDIO_CODEC_LOG_LEVEL_INHERIT` is selected. So if you select another value it should override the parent value.

theob-pro replied:
The other logging symbols uses `Kconfig.template.log_config_bt` and so have `BT` as `parent-module`. If we want to be consistent it could be nice to use it too for new one as I have done it in the upper example.

Thalley replied:
Done

MariuszSkamra commented on subsys/bluetooth/audio/Kconfig.cap:
```diff
@@ -48,7 +48,13 @@ config BT_CAP_INITIATOR
 
 config BT_DEBUG_CAP_INITIATOR
 	bool "Common Audio Profile Initiator debug"
+	select DEPRECATED
```
Is it needed to select `DEPRECATED` if the parent of Kconfig.cap - `BT_AUDIO` is marked as `EXPERIMENTAL`?

Thalley replied:
That is a good question. It is what had been done in https://github.com/zephyrproject-rtos/zephyr/pull/51897 for all LE Audio implementations, but I agree that the `deprecated` part could/should have been omitted for EXPERIMENTAL features.

Thalley committed:
Bluetooth: Audio: Audio ISO use log_config_inherit

Change the BT_AUDIO_ISO logging module to use
log_config_inherit to be consistent with the other LE Audio
logging modules.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>
